### ISSUE TITLE
Error: `make` failed with exit code: 2

### ISSUE THREAD

**Issue by funkyLover** [node-inspector/node-inspector#753]
node4.1.0
OS X 10.10.5
run `sudo npm install -g node-inspector` get the error
```
gyp WARN EACCES user "root" does not have permission to access the dev dir "/Users/funkyLover/.node-gyp/4.1.0"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/local/lib/node_modules/node-inspector/node_modules/ws/node_modules/utf-8-validate/.node-gyp"
  CXX(target) Release/obj.target/validation/src/validation.o
  SOLINK_MODULE(target) Release/validation.node
npm WARN engine xmlbuilder@2.2.1: wanted: {"node":"0.8.x || 0.10.x"} (current: {"node":"4.1.0","npm":"2.14.3"})

> bufferutil@1.2.1 install /usr/local/lib/node_modules/node-inspector/node_modules/ws/node_modules/bufferutil
> node-gyp rebuild

gyp WARN EACCES user "root" does not have permission to access the dev dir "/Users/funkyLover/.node-gyp/4.1.0"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/local/lib/node_modules/node-inspector/node_modules/ws/node_modules/bufferutil/.node-gyp"
  CXX(target) Release/obj.target/bufferutil/src/bufferutil.o
  SOLINK_MODULE(target) Release/bufferutil.node
\
> v8-debug@0.5.4 install /usr/local/lib/node_modules/node-inspector/node_modules/v8-debug
> node-pre-gyp install --fallback-to-build

[v8-debug] Success: "/usr/local/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.5.4/node-v46-darwin-x64/debug.node" is installed via remote

> v8-profiler@5.3.2 install /usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler
> node-pre-gyp install --fallback-to-build

gyp WARN EACCES user "root" does not have permission to access the dev dir "/Users/funkyLover/.node-gyp/4.1.0"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler/.node-gyp"
make: *** No rule to make target `../.node-gyp/4.1.0/include/node/common.gypi', needed by `Makefile'.  Stop.
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 14.5.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler/build/profiler/v5.3.2/node-v46-darwin-x64/profiler.node" "--module_name=profiler" "--module_path=/usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler/build/profiler/v5.3.2/node-v46-darwin-x64"
gyp ERR! cwd /usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler
gyp ERR! node -v v4.1.0
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler/build/profiler/v5.3.2/node-v46-darwin-x64/profiler.node --module_name=profiler --module_path=/usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler/build/profiler/v5.3.2/node-v46-darwin-x64' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:87:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:817:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
node-pre-gyp ERR! System Darwin 14.5.0
node-pre-gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler
node-pre-gyp ERR! node -v v4.1.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.10
node-pre-gyp ERR! not ok
Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler/build/profiler/v5.3.2/node-v46-darwin-x64/profiler.node --module_name=profiler --module_path=/usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler/build/profiler/v5.3.2/node-v46-darwin-x64' (1)
npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "-g" "node-inspector"
npm ERR! node v4.1.0
npm ERR! npm  v2.14.3
npm ERR! code ELIFECYCLE

npm ERR! v8-profiler@5.3.2 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the v8-profiler@5.3.2 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! This is most likely a problem with the v8-profiler package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get their info via:
npm ERR!     npm owner ls v8-profiler
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/funkyLover/npm-debug.log
```

**Comment by 3y3** [comment]
please try `sudo npm i -g node-inspector --unsafe-perm`

**Comment by funkyLover** [comment]
I can't figure out why
I tried the same `sudo npm install -g node-inspector` today
and it work well *_*.....

**Comment by WarFox** [comment]
I had similar issue with `npm install java` on Centos 6.4... related to `node-gyp rebuild`

Got it resolved by using `python27`, instead of `python 2.6.6`. Thought of sharing it here, if it helps some one.

@funkyLover did you have to upgrade python by any chance?

**Comment by funkyLover** [comment]
@WarFox 
No, I am sure I didn't upgrade pyhton or make other similar operation,

**Comment by WarFox** [comment]
Sorry, my bad. Apparently I had upgraded `gcc` as well along with `Python` on Centos. My gcc was below 4.8, before upgrading. Turns out this is real reason, why my problem got resolved.

**Comment by krainet** [comment]
I'm solved with: apt-get install libkrb5-dev

**Comment by amirlarco** [comment]
Try this may help you:

https://docs.npmjs.com/getting-started/fixing-npm-permissions

**Comment by borislemke** [comment]
Here's my case what fixed it:

I used `nvm` and had 7.2.1 installed.
1. Removed nvm `brew uninstall nvm`
2. Surely remove `node`, `sudo rm -rf /usr/local/{lib/node{,/.npm,_modules},bin,share/man}/{npm*,node*,man1/node*}`
3. Install node@6.9.2 `brew install node@6`
Errors gone!

**Comment by dave-maude** [comment]
I've just had this same issue on Centos 6.5. I managed to resolve it by installing a new version of GCC v6.2.1. I'm not 100% sure what this may equate to on a Mac, but my best guess would be that you need to update your XCode Command Line Tools which includes GCC for the Mac.

**Comment by laugh26** [comment]
am getting this problem when trying to install NOMP using npm update

make: *** [Release/obj.target/multihashing/multihashing.o] Error 1
make: Leaving directory '/home/pool/nomp/node_modules/multi-hashing/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:12)
gyp ERR! System Linux 4.10.0-38-generic
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/pool/nomp/node_modules/multi-hashing
gyp ERR! node -v v4.8.5
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! multi-hashing@0.0.9 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the multi-hashing@0.0.9 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2017-11-05T12_50_07_249Z-debug.log

**Comment by ajalbani1** [comment]
it happened to me when I had an npm package in es6 and I had my babel transpile ignored for node_modules.

Fixed it by importing the transpiled min.js that came with the package.

**Comment by iqasim-com** [comment]
It worked after deleting **package-lock.json** and re run npm install

**Comment by s1awek** [comment]
There is similar thread here: https://github.com/nodejs/node-gyp/issues/809
tldr;

> This has done the job
> Node 12.1.0, npm 6.9.0
> Remove node-gyp
> Update to node-sass 4.12.0
> npm update
> npm audit fix --force

**Comment by shaikh-shahid** [comment]
Follow these steps to resolve the issue.

1: Make sure you have build-tools installed. 
2: Update node-gyp 
3: Delete package-lock.json and node modules folder and run npm install again. 
4: Deleting ~/.node-gyp folder and run npm install again.

Source: https://codeforgeek.com/make-failed-with-exit-code-2/

**Comment by water-in-stone** [comment]
> I can't figure out why
> I tried the same `sudo npm install -g node-inspector` today
> and it work well ___.....

try `npm install -g node-gyp`
